test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+            sample_identity_count_type.get_type_name()));
+
+    ret = statistics_participant->enable_statistics_datawriter(
+            PHYSICAL_DATA_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PHYSICAL_DATA_TOPIC));
+    EXPECT_EQ(physical_data_type, statistics_participant->find_type(physical_data_type.get_type_name()));
+
+    // 5. Enable an enabled statistics DataWriter
+    ret = statistics_participant->enable_statistics_datawriter(
+            SAMPLE_DATAS_TOPIC, STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+
+    // 6. Invalid topic name
+    ret = statistics_participant->enable_statistics_datawriter(
+            "INVALID_TOPIC", STATISTICS_DATAWRITER_QOS);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription("INVALID_TOPIC"));
+
+    // 7. Disable statistics DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_NE(nullptr, statistics_participant->lookup_topicdescription(PDP_PACKETS_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            history_latency_type.get_type_name()));
+
+    // 8. Enable previous statistics DataWriter with an inconsistent QoS
+    ret = statistics_participant->enable_statistics_datawriter(
+            HISTORY_LATENCY_TOPIC, inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_INCONSISTENT_POLICY);
+    EXPECT_EQ(nullptr, statistics_participant->lookup_topicdescription(HISTORY_LATENCY_TOPIC));
+    EXPECT_EQ(eprosima::fastdds::dds::TypeSupport(nullptr), statistics_participant->find_type(
+            history_latency_type.get_type_name()));
+
+    // RETCODE_BAD_PARAMETER error has precedence over RETCODE_INCONSISTENT_POLICY
+    ret = statistics_participant->enable_statistics_datawriter(
+            "INVALID_TOPIC", inconsistent_qos);
+    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_BAD_PARAMETER);
+
+    // 9. Try to delete DomainParticipant and check that it fails because statistics DataWriters are still enabled
+    EXPECT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            delete_participant(statistics_participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_PRECONDITION_NOT_MET);
+
+    // 10. Disable already disabled DataWriter
+    ret = statistics_participant->disable_statistics_datawriter(HISTORY_LATENCY_TOPIC);
+    ASSERT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

Comment:
```suggestion
    EXPECT_EQ(ret, eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

Reply:
Done in b295b5fd0a602a3075552ff51925ccfb8afa32b3